packages/contracts-bedrock/src/L1/StandardValidator.sol

```diff
@@ -753,12 +774,56 @@ contract StandardValidator is ISemver {
+        // No super games and ONLY (permissionedDisputeGame) OR (permissionedDisputeGame + faultDisputeGame).
+        IDisputeGameFactory _factory = IDisputeGameFactory(_input.sysCfg.disputeGameFactory());
+        address _superCannon = address(_factory.gameImpls(GameTypes.SUPER_CANNON));
+        address _superPermissionedCannon = address(_factory.gameImpls(GameTypes.SUPER_PERMISSIONED_CANNON));
+        if (_superCannon == address(0) && _superPermissionedCannon == address(0)) {
+            _errors = assertValidPermissionedDisputeGame(
+                _errors, _input.sysCfg, _input.absolutePrestate, _input.l2ChainID, _input.proxyAdmin, false, _overrides
+            );
+            _errors = assertValidPermissionlessDisputeGame(
+                _errors, _input.sysCfg, _input.absolutePrestate, _input.l2ChainID, _input.proxyAdmin, false, _overrides
+            );
+
+            // Ensure respectedGameTypes matches one of the registered impls in each state.
+            GameType _respectedGameType =
+                IPermissionedDisputeGame(_superCannon).anchorStateRegistry().respectedGameType();
+            _errors = internalRequire(
+                GameType.unwrap(_respectedGameType) == GameType.unwrap(GameTypes.SUPER_CANNON)
+                    || GameType.unwrap(_respectedGameType) == GameType.unwrap(GameTypes.SUPER_PERMISSIONED_CANNON),
+                "GAMES-10",
+                _errors
+            );
+        } else if (_superCannon != address(0) && _superPermissionedCannon != address(0)) {
+            address _cannon = address(_factory.gameImpls(GameTypes.CANNON));
+            address _permissionedCannon = address(_factory.gameImpls(GameTypes.PERMISSIONED_CANNON));
+
+            _errors = assertValidPermissionedDisputeGame(
+                _errors, _input.sysCfg, _input.absolutePrestate, _input.l2ChainID, _input.proxyAdmin, true, _overrides
+            );
+            _errors = assertValidPermissionlessDisputeGame(
+                _errors, _input.sysCfg, _input.absolutePrestate, _input.l2ChainID, _input.proxyAdmin, true, _overrides
+            );
+
+            if (_cannon != address(0) && _permissionedCannon != address(0)) {
+                // Ensure respectedGameTypes matches one of the registered impls in each state.
+                address _game = _cannon == address(0) ? _permissionedCannon : _cannon;
+                GameType _respectedGameType = IPermissionedDisputeGame(_game).anchorStateRegistry().respectedGameType();
+                _errors = internalRequire(
+                    GameType.unwrap(_respectedGameType)
+                        == GameType.unwrap(_cannon == address(0) ? GameTypes.PERMISSIONED_CANNON : GameTypes.CANNON),
+                    "GAMES-20",
+                    _errors
+                );
+            }
+        } else {
+            // Only one super game is registered.
+            _errors = internalRequire(false, "GAMES-30", _errors);
+        }
+
```

Comment:
This if/else branch is quite complex. Can we try to contain this new logic into a small number of new internal functions? 

One option would be to first just check which implementations are set and which games are respected, and return some enum value, then based on that value we can decide which set of games need to be validated.